pkg/registry/servicecatalog/broker/strategy.go

```diff
@@ -148,6 +149,26 @@ func (brokerStatusRESTStrategy) PrepareForUpdate(ctx genericapirequest.Context,
 	}
 	// status changes are not allowed to update spec
 	newBroker.Spec = oldBroker.Spec
+
+	foundReadyConditionTrue := false
+	for _, condition := range newBroker.Status.Conditions {
+		if condition.Type == sc.BrokerConditionReady && condition.Status == sc.ConditionTrue {
+			foundReadyConditionTrue = true
+			break
+		}
+	}
+	if foundReadyConditionTrue {
```

Comment:
Minor thing but if we "break" upon first match and we "return" in this if-statement, we don't really need "foundReadyConditonTrue" - just move this if-logic up above and get rid of the variable.